Yolo skeleton code

Nathan903 opened the pull request.

Nathan903 committed:
upload yolov7 human tracker

Nathan903 committed:
Update readme.md

Nathan903 committed:
fixed fps display and removed unused code

stevenwaslander commented on human_detection/yolov7 skeleton/utils/google_app_engine/Dockerfile:
Are you using this dockerfile for training?

Nathan903 replied:
No. These are not used. The `models/` and `utils/` folder are downloaded as is from the yolov7 official repo without any modification. I can go through to remove unnecessary files if it would make the code better.

stevenwaslander commented on human_detection/yolov7 skeleton/utils/aws/__init__.py:
Are you training on AWS? We have our own servers to train on.

Nathan903 replied:
(same as above) I didn't do any training. The pretrained model is downloaded from the official repo.
I can go through to remove unnecessary files if it would make the code better.

stevenwaslander commented on human_detection/yolov7 skeleton/main.py:
Is this your code? If so, I'll look through it a bit more carefully.

Nathan903 replied:
Yes. I have renamed it to `yolov7.py` 
here is the list of files I wrote that important:
 - `yolov7.py` code to run the model
 - `example_usage.py` code to show how to call the yolov7.py in other codes
 - `human_detection_node.py` ros2 node that uses yolov7.py

Nathan903 committed:
added default options

so command line opt is no longer required to run model

also cleaned up some confusing variable names

Nathan903 committed:
rename main.py to yolov7.py

Nathan903 committed:
initialize sort tracker in the constructor

Nathan903 committed:
added more comments for the options

Nathan903 committed:
output tracking identities and confidences

Nathan903 committed:
added example_usage.py

Nathan903 committed:
Create human_detection_node.py

Nathan903 committed:
changed movenet to yolov7

Nathan903 commented on human_detection/yolov7 skeleton/.gitignore:
> should this be here? 
this .gitignore is to ignore the model weight and video files. Should I move this to the root folder's .ignore file?

> and no space in folder name.
got it, i will rename with underscore

Nathan903 committed:
Removing globals in human detection (#50)

* remove all globals

* moved all constants to yaml file

* fixed typo and added ApproachState

Nathan903 committed:
Update example_usage.py

Nathan903 committed:
Delete human_detection_node.py

Nathan903 committed:
rename folder to remove space

Nathan903 committed:
Update readme.md

Nathan903 committed:
Update readme.md

Nathan903 committed:
fixed global var error

Nathan903 committed:
copy human_detection_node to yolov7

Nathan903 committed:
change movenet to yolov7

Nathan903 committed:
remove extra dependencies

Nathan903 committed:
import vision_msgs

Nathan903 committed:
Update human_detection_node.py

untested

Nathan903 committed:
Update readme.md

Nathan903 committed:
added model file existence check

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
d

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
a

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
updated readme

stevenwaslander commented on human_detection/human_detection/human_detection_node.py:
```diff
@@ -443,28 +470,29 @@ def estimate_depth(x, y, np_2d_array):
-    debug_mode = True
     print("\n\nDEBUG MODE ON\n\n")
-    main()
+    main(debug_mode=True)
```
permanently in debug mode?

Nathan903 replied:
no. Debug mode is on when the someone runs the python file by itself, and off if ran as a ros node

stevenwaslander commented on human_detection/yolov7_skeleton/human_detection_node.py:
```diff
@@ -0,0 +1,562 @@
+        self.create_timer(1/self.publishing_frequency, self.publish_message)
+        self.print_and_log('Human Detection ready...')
+        
+        ascii_numbers = r"""
```
Do you really need this three times in the same PR?!

stevenwaslander replied:
That's a lot of identical count downs.

nickoenig37 committed:
moved yolo to gpu, fixed memory leak, added init.py to yolov7_skeleton

Nathan903 committed:
remove old code

Nathan903 committed:
move yolov7 version to human_detection node

Nathan903 committed:
change import to relative

Nathan903 committed:
Update human_detection_node.py

Nathan903 committed:
Update yolov7.py

Nathan903 committed:
Update yolov7.py

Nathan903 committed:
Update yolov7.py

Nathan903 committed:
Update google_utils.py

nickoenig37 committed:
new launch file

stevenwaslander commented on trailbot_bringup/launch/nav_camera.launch.py:
Remove, not relevant

stevenwaslander replied:
This is ok, wrong branch, but still useful.  A change from trailbot testing.

stevenwaslander commented on human_detection/human_detection/utils/wandb_logging/__init__.py:
I don't think we need this folder, we're not doing any training or w and b logging.

stevenwaslander commented on human_detection/human_detection/utils/google_app_engine/Dockerfile:
I don't think we need this folder, we're not doing any google_app engine stuff.

stevenwaslander commented on human_detection/human_detection/utils/aws/__init__.py:
I don't think we need this folder, we're not doing any aws stuff.

stevenwaslander reviewed (commented):
Just some minor cleanup if possible today.  Will test on trailbot as is.

Nathan903 committed:
Update human_detection_node.py
